### PR TITLE
Add completion for DebugConfig name and path arguments

### LINKED ISSUE #43594: [DX] Add completion to Symfony commands
Completion of command arguments and options is coming in Symfony 5.4 (#38275). Command and option names are already autocompleted, but argument and option values needs to be implemented on each command.

As an example, [#42251](https://github.com/symfony/symfony/pull/42251/files#diff-771e524eaa21e31d32047fda540ba0b53222c74e693bee5a45f6da7fa1644571R87-R96) added completion to  `secrets:remove`.

This feature needs to be implemented on Symfony commands that has predictable values for their arguments and options.

Some guidelines:

- Don't complete file/directory paths for now. See https://github.com/symfony/symfony/issues/43594#issuecomment-947385485
- You don't need to filter the suggestions in PHP, `compgen` (used by the shell script) does this for you based on the suggestions provided by PHP
- You can use the debugging utility while working on providing suggestions. See https://github.com/symfony/symfony/issues/43594#issuecomment-947535113

**PLEASE, comment here BEFORE starting to work on something and WAIT for the confirmation to avoid duplicate work.**

*All commands have been claimed*

 * [x] Command `debug:event-dispatcher` (`event`, `--dispatcher`, `--format`) @stephenkhoo https://github.com/symfony/symfony/pull/43859
 * [x] Command `debug:config` (`name`, `path`) @eclairia https://github.com/symfony/symfony/pull/43850
 * [x] Command `debug:autowiring` (`search`) @eclairia https://github.com/symfony/symfony/pull/43838
 * [x] Command `messenger:failed:remove` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `messenger:failed:retry` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `messenger:failed:show` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `debug:container` (`name`) @GromNaN https://github.com/symfony/symfony/pull/43857
 * [x] Command `messenger:consume` (`receivers`, `--bus`, `--queues`) @GromNaN https://github.com/symfony/symfony/pull/43891
 * [x] Command `debug:firewall` (`name`, `--format`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:form` (`name`, `--format`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:messenger` (`bus`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:router` (`name`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `cache:pool:clear` (`pools`) @andyexeter https://github.com/symfony/symfony/pull/43621
 * [x] Command `cache:pool:delete` (`pool`, `key`) @andyexeter https://github.com/symfony/symfony/pull/43621
 * [x] Command `completion` (`shell`) @dkarlovi https://github.com/symfony/symfony/pull/43615
 * [x] Command `config:dump-reference` (`name`, `path`) @StaffNowa https://github.com/symfony/symfony/pull/43682
 * [x] Command `help` (`command_name`, `--format`) @GromNaN https://github.com/symfony/symfony/pull/43596
 * [x] Command `debug:translation` (`locale`, `bundle`, `--domain`) @alexandre-daubois https://github.com/symfony/symfony/pull/43644
 * [x] Command `debug:twig` (`name`, `--filter`, `--format`) @StaffNowa https://github.com/symfony/symfony/pull/43846
 * [x] Command `list` (`namespace`, `--format`) @GromNaN https://github.com/symfony/symfony/pull/43596
 * [x] Command `lint:twig` (`filename`, `--format`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `lint:xliff` (`filename`, `--format`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `lint:yaml` (`filename`, `--format`, `--exclude`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `messenger:setup-transports` (`transport`) @Tayfun74 https://github.com/symfony/symfony/pull/43640
 * [x] Command `secrets:remove` (`name`) @GromNaN https://github.com/symfony/symfony/pull/43626
 * [x] Command `secrets:set` (`name`, `file`) @GromNaN https://github.com/symfony/symfony/pull/43626
 * [x] Command `security:encode-password` (`user-class`) @94noni https://github.com/symfony/symfony/pull/43653
 * [x] Command `security:hash-password` (`user-class`) @94noni https://github.com/symfony/symfony/pull/43653
 * [x] Command `server:dump` (`--format`) @alexandre-daubois https://github.com/symfony/symfony/pull/43683
 * [x] Command `translation:pull` (`provider`, `--domains`, `--locales`, `--format`) @welcoMattic https://github.com/symfony/symfony/pull/43672 
 * [x] Command `translation:push` (`provider`, `--domains`, `--locales`) @welcoMattic https://github.com/symfony/symfony/pull/43672 
 * [x] Command `translation:update` (`locale`, `bundle`, `--format`, `--domain`, `--sort`) @stephenkhoo https://github.com/symfony/symfony/pull/43676
 * [x] Command `ulid:generate` (`--format`) @StaffNowa https://github.com/symfony/symfony/pull/43639
 * [x] Command `uuid:generate` (`--format`) @StaffNowa https://github.com/symfony/symfony/pull/43639
 * [x] Command `workflow:dump` (`name`, `marking`, `--label`, `--format`) @StaffNowa StaffNowa https://github.com/symfony/symfony/pull/43848

Commands that does not require auto-completion:

* [x] Command `about`
* [x] Command `assets:install` (`target`) (directory completion)
* [x] Command `cache:clear`
* [x] Command `cache:pool:list`
* [x] Command `cache:pool:prune`
* [x] Command `cache:warmup`
* [x] Command `debug:dotenv`
* [x] Command `dotenv:dump`
 * [x] Command `debug:validator` (`class`) @Kocal *completion support not yet possible*
* [x] Command `lint:container`
* [x] Command `messenger:stop-workers`
* [x] Command `router:match` (`path_info`)
* [x] Command `secrets:decrypt-to-local`
* [x] Command `secrets:encrypt-from-local`
* [x] Command `secrets:generate-keys`
* [x] Command `secrets:list`
* [x] Command `server:log` (`--host`, `--format`) @StaffNowa https://github.com/symfony/symfony/pull/43892
* [x] Command `ulid:inspect`
* [x] Command `uuid:inspect`

### PR DESCRIPTION
**Pull request by eclairia**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #43594
| License       | MIT
| Doc PR        | -

Adding Bash completion for debug:config command.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php [hunk 1]
```diff
@@ -188,4 +186,53 @@ private function getConfigForExtension(ExtensionInterface $extension, ContainerB
...
+    }
+
+    private function getBundlesCompletion(string $name): array
+    {
+        $bundles = [];
+        foreach ($this->getApplication()->getKernel()->getBundles() as $bundle) {
+            $bundleName = $bundle->getName();
+            $bundleAlias = $bundle->getContainerExtension()->getAlias();
+
+            if (str_starts_with(strtolower($bundleName), $name)) {
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: Filtering suggestions is not necessary. The bash completion tool will do it. You can return the full list of bundles + aliases, like in command `debug:translation`. https://github.com/symfony/symfony/pull/43644/files#diff-67afa5b8860d0df4e44f1e1fc89f444b7ac77de515b698a6824dd5403a0acdbcR270-R283

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php [hunk 2]
```diff
@@ -188,4 +186,53 @@ private function getConfigForExtension(ExtensionInterface $extension, ContainerB
...
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        $name = strtolower($input->getArgument('name'));
+
+        if ($input->mustSuggestArgumentValuesFor('name')) {
+            $suggestions->suggestValues($this->getBundlesCompletion($name));
+        }
+
+        if ($input->mustSuggestArgumentValuesFor('path') && null !== $name) {
+            $path = $input->getArgument('path');
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: The `path` argument value cannot be read when it is the current completed value.

**Review by GromNaN — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php (hunk 1 above)
Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I replaced my code with the logic of the `debug:translation` command.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php (hunk 2 above)
Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I added an early return to the `name` argument to avoid reading `path` argument before it is completed.

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I think it still cannot work. Here, we are trying to autocomplete the path, so there is no way we can get the path value.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php (hunk 2 above)
Comment (GromNaN) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: This is a path in config tree. The exhaustive list of values can be generated.

Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I reworked my way to autocomplete path argument.
Now I get the path tree thanks to `getConfigForExtension` and build a string path recursively for autocompletion.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php [hunk 3]
```diff
@@ -190,36 +185,26 @@ private function getConfigForExtension(ExtensionInterface $extension, ContainerB
     public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
     {
         if ($input->mustSuggestArgumentValuesFor('name')) {
-            $suggestions->suggestValues($this->getAvailableBundles());
+            $suggestions->suggestValues($this->getAvailableBundles(!preg_match('/[A-Z]/', $input->getCompletionValue()[0] ?? '')));
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: That avoids suggesting twice the each bundle with syntax (`FrameworkBundle` and `framework`) and switch to the must accurate suggestions when something has been typed.

```suggestion
            $suggestions->suggestValues($this->getAvailableBundles(!preg_match('/^[A-Z]/', $input->getCompletionValue())));
```

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDebugCommandTest.php [hunk 4]
```diff
@@ -122,41 +122,36 @@ public function testComplete(array $input, array $expectedSuggestions)
...
-                'DefaultConfigTestBundle',
                 'default_config_test',
-                'ExtensionWithoutConfigTestBundle',
                 'extension_without_config_test',
-                'FrameworkBundle',
                 'framework',
-                'TestBundle',
                 'test',
             ],
         ];
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDebugCommandTest.php]: You can add the test for BundleName.

```suggestion

        yield 'name (started CamelCase)' => [['Fra'], ['DefaultConfigTestBundle', 'ExtensionWithoutConfigTestBundle', 'FrameworkBundle', 'TestBundle']];
```

**Review by GromNaN — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php (hunk 3 above)
Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I replaced my code by your suggestion @GromNaN.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDebugCommandTest.php (hunk 4 above)
Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDebugCommandTest.php]: I added your test suggestion @GromNaN.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php [hunk 5]
```diff
@@ -50,16 +52,11 @@ protected function configure()
             ->setHelp(<<<'EOF'
 The <info>%command.name%</info> command dumps the current configuration for an
 extension/bundle.
-
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: Can you revert all unrelated changes?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php (hunk 5 above)
Comment (eclairia) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php]: I reverted all unrelated changes @fabpot.

**Review by GromNaN — APPROVED** [review]
Test failures are not related to the changes.

**Comment by fabpot** [comment]
Thank you @eclairia.
